www/docs/guides/migrate_v7_v8.md

```diff
@@ -0,0 +1,536 @@
+This affects how you handle transaction states and block confirmations in your applications.
+
+## Overview
+
+Starknet.js v8 introduces several breaking changes while adding full support for these Starknet 0.14 protocol changes and RPC 0.9. The most significant change is the move from argument-based constructors to object-based APIs for better developer experience and extensibility.
+
+## Node.js Requirements
+
+Starknet.js v8 has a soft requirement for **Node.js >= 22** due to built-in WebSocket support.
+
+- **Node.js >= 22**: WebSocket support is built-in, no additional dependencies needed
+- **Node.js < 22**: You can still use older versions but will need to provide a WebSocket implementation
+
+For older Node.js versions, install the `ws` library and configure it:
+
+```bash
+npm install ws
+npm install --save-dev @types/ws
+```
+
+```typescript
+import { WebSocketChannel, config } from 'starknet';
+import WebSocket from 'ws';
+
+// Option 1: Configure WebSocket globally using config singleton
+config.set('websocket', WebSocket);
+
+// Option 2: Pass WebSocket to WebSocketChannel constructor
+const channel = new WebSocketChannel({
+  nodeUrl: 'wss://...',
+  websocket: WebSocket,
+});
+```
+
+## RPC Version Support
+
+Starknet.js v8 introduces support for **RPC 0.9** and maintains compatibility with **RPC 0.8**. **RPC 0.7 support has been removed.**
+
+|                       |   RPC 0.7   |    RPC 0.8    | RPC 0.9 <br /> (new) |
+| --------------------: | :---------: | :-----------: | :------------------: |
+| V3 tx (Starknet fees) |     ❌      |       ✓       |          ✓           |
+|    **Support Status** | **Removed** | **Supported** |    **Supported**     |
+
+## Breaking Changes
+
+### RPC 0.7 Support Removed
+
+**Important:** Starknet.js v8 no longer supports RPC 0.7. If you were using RPC 0.7 endpoints, you must upgrade to RPC 0.8 or RPC 0.9.
+
+**v8 (use RPC 0.8 or 0.9):**
+
+```typescript
+// Option 1: Use RPC 0.8
+const provider = new RpcProvider({
+  nodeUrl: 'https://starknet-sepolia.public.blastapi.io/rpc/v0_8',
+  specVersion: '0.8.1',
+});
+
+// Option 2: Use RPC 0.9 (default)
+const provider = new RpcProvider({
+  nodeUrl: 'https://starknet-sepolia.public.blastapi.io/rpc/v0_9',
+  // specVersion defaults to '0.9.0'
+});
+```
+
+### Constructor API Changes
+
+All major classes now use object-based constructors instead of positional arguments for better clarity and extensibility.
+
+#### Account Class
+
+**v7 (Arguments-based):**
+
+```typescript
+import { Account, RpcProvider } from 'starknet';
+
+const provider = new RpcProvider({
+  nodeUrl: 'https://starknet-sepolia.public.blastapi.io/rpc/v0_7',
+});
+const account = new Account(
+  provider,
+  accountAddress,
+  privateKey,
+  undefined,
+  ETransactionVersion.V3
+);
+```
+
+**v8 (Object-based):**
+
+```typescript
+import { Account, RpcProvider } from 'starknet';
+
+const provider = new RpcProvider({
+  nodeUrl: 'https://starknet-sepolia.public.blastapi.io/rpc/v0_7',
```

Comment:
v8 example, but the node URL is set to v7. The `import` and `provider` initialization could be omitted and just the `account` initialization kept.